kvcache/kvcache.go

```diff
@@ -33,23 +37,23 @@ func (c *SimpleKeyValueCache) Create(key,value string) error{
 	if _, ok := c.Data[key]; ok {
 		return fmt.Errorf("create failed: key '%v' isn't unique: ", key)
 	}
-	c.Data[key] = value
 
+	c.Data[key] = value
 	return nil
 }
 
 //updated interface and method to return both string and error when realized SKVC wouldn't return an error when an empty string was entered as a key - not cool
 func (c *SimpleKeyValueCache) Read(key string) (string,error){
 	err := c.Data[key]
```

Comment:
Two things:
* Reserve `err` for error types. 
* If you use two values for the output of map access, you will get the value and a boolean. You can use this boolean to check if the key exists in the cache instead of checking empty string.